CRM/Price/BAO/PriceSet.php

```diff
@@ -827,8 +827,12 @@ public static function buildPriceSet(&$form) {
       $validFieldsOnly = FALSE;
     }
 
-    $priceSet = self::getSetDetail($priceSetId, TRUE, $validFieldsOnly);
-    $form->_priceSet = $priceSet[$priceSetId] ?? NULL;
+    // Don't build priceSet multiple times. If called via a contributionPage it was already built by
+    // CRM_Price_BAO_PriceSet::initSet()
+    if (!$form->_priceSet) {
```

Comment:
Side note - I've been moving towards $form->getPriceSet() rather than $form->_priceSet because it's easier to start to make it a bit smarter over time (not necessarily appropriate / do-able here - just a general comment)